cmd/libsnap-confine-private/tool.c

```diff
@@ -117,6 +117,22 @@ void sc_call_snap_update_ns_as_user(int snap_update_ns_fd,
 				 "XDG_RUNTIME_DIR=%s", xdg_runtime_dir);
 	}
 
+	const char *snap_real_home = getenv("SNAP_REAL_HOME");
+	char snap_real_home_env[PATH_MAX + sizeof("SNAP_REAL_HOME=")] = { 0 };
```

Comment:
The spec specifically says "Sanity check the trustworthiness of $SNAP_REAL_HOME by ensuring $SNAP_UID == actual uid" - but I don't see that implemented here. I agree with @pedronis though that we need to be very careful here since anyway can launch snap-confine with crafted values for SNAP_REAL_HOME and SNAP_UID - and hence these both need to be carefully validated.

Reply:
I provided an updated proposal in [this comment.](https://github.com/snapcore/snapd/pull/13244#issuecomment-1775303753). My proposal is to make verification of the passed on env vars the responsibility of the consumer `snap-update-ns`.

Fwiw, the sanity check mentioned in the spec was intended for `snap-update-ns`.